addons/mail/tests/common.py

```diff
@@ -1259,3 +1260,59 @@ def _generate_attachments_data(count, res_model, res_id, attach_values=None, pre
             'res_id': res_id,
             **attach_values,
         } for x in range(count)]
+
+class MockStatusbarTime(MailCommon):
```

Comment:
Have to say this is not crystal clear ... issue for me are that you call "assert" a method that is actually doing some updates of data, computing stuff then asserting other stuff. Moreover it relies on variables on "self" that are not really clear... you cannot use this tool without defining "self.rec_1" for example.

Moreover it is not really a mock, as you don't mock anything. Can't you somehow merge a bit MockStatusbarTime and TestStatusbarTimeMixin to ease its integration and usage ?

I agree we need something to insert fake tracking, a tool to compute the difference of json between two update maybe, then an assert (assertTrackingDuration ?) that checks the content of the field. But currently everything it a bit mixed up, this could probably be cleaned a bit more :) .

Reply:
I refactored it and I think it's a lot clearer now but I'll let you be the judge!